lib/r10k/util/subprocess/runner/childprocess.rb

```diff
@@ -0,0 +1,45 @@
+require 'r10k/util/subprocess/runner'
+require 'r10k/util/subprocess/runner/pump'
+require 'childprocess'
+
+class R10K::Util::Subprocess::Runner::Childprocess < R10K::Util::Subprocess::Runner
+
+  def initialize(argv)
+    @argv = argv
+  end
+
+  def run
+    process = ChildProcess.build(*@argv)
+    process.cwd = @cwd if @cwd
+
+    stdout = ''
+    stderr = ''
+    exit_code = 254
+
+    stdout_r, stdout_w = IO.pipe
+    stdout_pump = R10K::Util::Subprocess::Runner::Pump.new(stdout_r)
+    stderr_r, stderr_w = IO.pipe
+    stderr_pump = R10K::Util::Subprocess::Runner::Pump.new(stderr_r)
+
+    process.io.stdout = stdout_w
+    process.io.stderr = stderr_w
+
+    begin
+      stdout_pump.start
+      stderr_pump.start
+      process.start
+      process.wait
+      stdout_pump.halt!
+      stderr_pump.halt!
```

Comment:
Good catch! I was relying on the garbage collector to close the file descriptors when the IO objects go out of scope but since that's proven to be problematic I'll throw in the explicit closes.